### PR TITLE
Made IntermediateNodeWalker.Ancestors indexable

### PR DESCRIPTION
**Pull request by ajaybhargavb**
#1510 

@rynowak @NTaylorMullen

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs [hunk 1]
```diff
@@ -7,11 +7,11 @@ namespace Microsoft.AspNetCore.Razor.Language.Intermediate
 {
     public abstract class IntermediateNodeWalker : IntermediateNodeVisitor
     {
-        private readonly Stack<IntermediateNode> _ancestors = new Stack<IntermediateNode>();
+        private readonly List<IntermediateNode> _ancestors = new List<IntermediateNode>();
 
-        protected IEnumerable<IntermediateNode> Ancestors => _ancestors;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs]: Unit test?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs]: There is already a unit test that verifies if the ancestors are set correctly. Did you mean add one that tests indexability?

Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs]: Does it verify the iteration order? If it does then consider me signed off.

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeWalker.cs]: Yes it does 👍 https://github.com/aspnet/Razor/blob/dev/test/Microsoft.AspNetCore.Razor.Language.Test/Intermediate/IntermediateNodeWalkerTest.cs#L68.

**Pull request merged**